pyne/source_sampling.pyx

```diff
@@ -267,48 +294,21 @@ cdef class Sampler:
         Returns
         -------
         None
-        
-        ################################################################
-        
-        Constuctor for analog and uniform sampling
-        
-        Parameters
-        ----------
-        e_bounds : std::vector< double >
-        
-        src_tag_name : std::string
-        
-        uniform : bool
-        
-        filename : std::string
-        
-        Returns
-        -------
-        None
-        
         """
         cdef char * filename_proxy
         cdef char * src_tag_name_proxy
-        cdef cpp_vector[double] e_bounds_proxy
-        cdef int ie_bounds
-        cdef int e_bounds_size
-        cdef double * e_bounds_data
         cdef char * bias_tag_name_proxy
         filename_bytes = filename.encode()
         src_tag_name_bytes = src_tag_name.encode()
-        # e_bounds is a ('vector', 'float64', 0)
-        e_bounds_size = len(e_bounds)
-        if isinstance(e_bounds, np.ndarray) and (<np.ndarray> e_bounds).descr.type_num == np.NPY_FLOAT64:
-            e_bounds_data = <double *> np.PyArray_DATA(<np.ndarray> e_bounds)
-            e_bounds_proxy = cpp_vector[double](<size_t> e_bounds_size)
-            for ie_bounds in range(e_bounds_size):
-                e_bounds_proxy[ie_bounds] = e_bounds_data[ie_bounds]
-        else:
-            e_bounds_proxy = cpp_vector[double](<size_t> e_bounds_size)
-            for ie_bounds in range(e_bounds_size):
-                e_bounds_proxy[ie_bounds] = <double> e_bounds[ie_bounds]
         bias_tag_name_bytes = bias_tag_name.encode()
-        self._inst = new cpp_source_sampling.Sampler(std_string(<char *> filename_bytes), std_string(<char *> src_tag_name_bytes), e_bounds_proxy, std_string(<char *> bias_tag_name_bytes))
+        # convert e_bounds
+        cdef cpp_vector[double] e_bounds_proxy = convert_e_bounds(e_bounds)
```

Comment:
But I thought it was being used here

Reply:
I am sorry. The function removed is `get_xyzew`, but not the `convert_e_bounds`. I mistook these two functions. By the way, the return type is cpp_vector[double], I have changed that in the updated documentation on `convert_e_bounds`.

Reply:
Ahh ok. Also, the return type **is not** `cpp_vector[double]` because Cython will automatically convert `cpp_vector[double]` to a Python list of Python floats. This is happening because it is `def` (Python) function rather than a `cdef` C/C++ function. See http://cython.readthedocs.io/en/latest/src/userguide/wrapping_CPlusPlus.html#standard-library

So what is happening in the code is that the vector is being returned as a list and then the list is being converted back to a vector at the call site. This is exactly what we want to avoid :smile:

Reply:
As this is the problem of the `def` and `cdef`. Does that means we can avoid that by define the `convert_e_bounds` as `cdef`?
```
cdef convert_e_bounds(e_bounds):
```

Reply:
yes, but you need to add the return type (and probably the argument type) too:

```cython
cdef cpp_vector[double] convert_e_bounds(e_bounds):
```

Reply:
Great thanks!